docker/marin/preemptible-fix.patch

```diff
@@ -0,0 +1,57 @@
+diff --git a/python/ray/autoscaler/_private/autoscaler.py b/python/ray/autoscaler/_private/autoscaler.py
+index eec513cea969..85b52aac2e82 100644
```

Comment:
Just to verbalize my understanding of this patch.

Ray has logic to schedule node death and to mark new nodes as alive. If the node is simply pre-empted, both of these code paths start erroring in a loop (presumably, Ray keeps trying to kill the "unhealthy" node but it can't actually get it).

Since schedule_node_termination doesn't error when getting the ip anymore, ray is now tricked into thinking it killed the node and everything is happier?

Reply:
i dunno i just copied it from Abhi's fork